tcms_django_plugin/result.py

```diff
@@ -32,26 +32,46 @@ def startTest(self, test):
 
     def addSuccess(self, test):
         super().addSuccess(test)
-        self.status_id = self.backend.get_status_id('PASSED')
+        try:
+            self.status_id = self.backend.get_status_id('PASSED')
+        except IndexError:
+            self.status_id = self.backend.rpc.TestExecutionStatus.filter(
+                {'weight__gt': 0})[-1]['id']
 
     def addError(self, test, err):
         super().addError(test, err)
-        self.status_id = self.backend.get_status_id('FAILED')
+        try:
+            self.status_id = self.backend.get_status_id('FAILED')
+        except IndexError:
+            self.status_id = self.backend.rpc.TestExecutionStatus.filter(
+                {'weight__lt': 0})[-1]['id']
         self.comment = self.errors[-1][1]
 
     def addFailure(self, test, err):
         super().addFailure(test, err)
-        self.status_id = self.backend.get_status_id('FAILED')
+        try:
+            self.status_id = self.backend.get_status_id('FAILED')
+        except IndexError:
+            self.status_id = self.backend.rpc.TestExecutionStatus.filter(
+                {'weight__lt': 0})[-1]['id']
         self.comment = self.failures[-1][1]
 
     def addSkip(self, test, reason):
         super().addSkip(test, reason)
-        self.status_id = self.backend.get_status_id('WAIVED')
+        try:
+            self.status_id = self.backend.get_status_id('WAIVED')
+        except IndexError:
+            self.status_id = self.backend.rpc.TestExecutionStatus.filter(
+                {'weight__gt': 0})[0]['id']
         self.comment = reason
 
     def addExpectedFailure(self, test, err):
         super().addExpectedFailure(test, err)
-        self.status_id = self.backend.get_status_id('FAILED')
+        try:
+            self.status_id = self.backend.get_status_id('FAILED')
+        except IndexError:
+            self.status_id = self.backend.rpc.TestExecutionStatus.filter(
+                {'weight__lt': 0})[-1]['id']
         self.comment = self.expectedFailures[-1][1]
 
     def addUnexpectedSuccess(self, test):
```

Comment:
This is missing the self.status_id integration and should report under FAILED because is a result which wasn't expected (the opposite of ExpectedFailure)

Reply:
^^^

Reply:
sorry, not sure how I had missed this.